### PR TITLE
fix: [PIE-4978]: name field not rendering

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

name field not rendering in input set form 
The filePath does not populate the first time a person visits an already created input set, but it does populate on a visual yaml switch.
#### Screenshots

NA

https://user-images.githubusercontent.com/106532291/185659341-607af454-d2a9-45a9-8ea4-c8e5b0ef5cce.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10656)

**Review by Eric761 — DISMISSED** [review]

**Review by YounixIO — DISMISSED** [review]

**Comment by vikrant-harness** [comment]
retrigger jest

**Comment by vikrant-harness** [comment]
retrigger coverage

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 1]
```diff
@@ -312,7 +312,7 @@ export default function FormikInputSetForm(props: FormikInputSetFormProps): Reac
 
   React.useEffect(() => {
     const initialValues = getPipelineData()
-    formikRef.current?.setFieldValue('pipeline', initialValues.pipeline)
+    formikRef.current?.setValues({ ...initialValues, ...storeMetadata })
   }, [inputSet, isEdit, resolvedPipeline])
```
Comment (kapilkumawat86) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:316]: @vkbansal-harness please check this change

**Review by kapilkumawat86 — COMMENTED** [review]
let's do proper testing before making this change

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx [hunk 2]
```diff
@@ -312,7 +312,7 @@ export default function FormikInputSetForm(props: FormikInputSetFormProps): Reac
 
   React.useEffect(() => {
     const initialValues = getPipelineData()
-    formikRef.current?.setFieldValue('pipeline', initialValues.pipeline)
+    formikRef.current?.setValues({ ...initialValues, ...storeMetadata })
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:315]: There's 1 more place where `formikRef.current.setFieldValue('pipeline', newPipeline)` is called. Is this fix not required there? Line 257

**Review by depanshu-dhiman — COMMENTED** [review]
Need to know if Line 257 does not need similar fix

File: src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:315]: @depanshu-dhiman line number 257 corresponds to CI , my changes are currently related to CD only will need to check with CI as well if they are facing the same issue as well.

Comment (depanshu-dhiman) [reply src/modules/70-pipeline/components/InputSetForm/FormikInputSetForm.tsx:315]: Yes, please check. I'll approve this for now.

**Review by depanshu-dhiman — APPROVED** [review]

**Pull request merged**